includes/options/views/class-amp-setup-wizard-submenu-page.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * AMP setup wizard page.
+ *
+ * @package AMP
+ * @since @todo NEW_ONBOARDING_RELEASE_VERSION
+ */
+
+/**
+ * AMP setup wizard submenu page class.
+ *
+ * @since @todo NEW_ONBOARDING_RELEASE_VERSION
+ */
+final class AMP_Setup_Wizard_Submenu_Page {
+	/**
+	 * Handle for JS file.
+	 *
+	 * @var string
+	 */
+	const JS_HANDLE = 'amp-setup';
+
+	/**
+	 * HTML ID for the app root element.
+	 *
+	 * @var string
+	 */
+	const APP_ROOT_ID = 'amp-setup';
+
+	/**
+	 * Sets up hooks.
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+	}
+
+	/**
+	 * Renders the setup screen markup.
+	 */
+	public function render() {
+		?>
+			<div id="<?php echo esc_attr( static::APP_ROOT_ID ); ?>"></div>
+		<?php
+	}
+
+	/**
+	 * Provides the setup screen handle.
+	 *
+	 * @return string
+	 */
+	public function screen_handle() {
+		return sprintf( 'amp_page_%s', AMP_Setup_Wizard_Submenu::SCREEN_ID );
+	}
+
+	/**
+	 * Enqueues setup assets.
+	 *
+	 * @param string $hook_suffix The current admin page.
+	 */
+	public function enqueue_assets( $hook_suffix ) {
+		if ( $this->screen_handle() !== $hook_suffix ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			self::JS_HANDLE,
+			amp_get_asset_url( 'js/' . self::JS_HANDLE . '.js' ),
+			[],
+			AMP__VERSION,
```

Comment:
Shouldn't this be using the `version` as exported into `amp-setup.assets.php`?

Reply:
Actually I'm not seeing an `amp-setup.asset.php` being generated like for the other scripts. Why?

Reply:
Good catch. The `DependencyExtractionWebpackPlugin` plugin is not being applied in the Webpack config. I'll make a suggestion.

Reply:
If we use the dependencies from the dependency extraction plugin here, this will by default end up enqueuing all the core packages that we'll now be bundling into our JS. So we need the dependency array to be empty. We can definitely use the generated version here, though, so I'll make that update with @pierlon's suggestion below.